File: server/src/types/vehicle.ts
export interface Vehicle {
  id?: string;
  vehicleNumber?: string;
  vehicleType?: string;
  capacity?: number;
  loadedFuelCostPerKm?: number;
  unloadedFuelCostPerKm?: number;
  currentLatitude?: number;
  currentLongitude?: number;
  landFillId?: string;
  duration?: number;
  distance?: number;
  createdAt?: Date;
  updatedAt?: Date;
  busyTime?: number;
  count?: number;
}

File: waste_management/lib/models/vehicle.dart
class Vehicle {
  String vehicleNumber;
  String vehicleType;
  int capacity;
  String landfillName;
  double unloadedCost;
  double loadedCost;

  Vehicle({
    required this.vehicleNumber,
    required this.vehicleType,
    required this.capacity,
    required this.landfillName,
    required this.unloadedCost,
    required this.loadedCost,
  });
}